parameter_api.cpp

```diff
@@ -9,30 +9,30 @@ static std::string trim(std::string s)
     return s.substr(first, (last-first+1));
 }
 
-void ParameterAPI::add_api_variable(std::string name, APIVariable *var) {
+void ParameterAPI::add_api_variable(const char * name, APIVariable *var) {
     variable_map_[name] = var;
     auto_complete_.add_match_string(name);
 }
 
-void ParameterAPI::add_api_variable(std::string name, const APIVariable *var) {
+void ParameterAPI::add_api_variable(const char * name, const APIVariable *var) {
     const_variable_map_[name] = var;
     auto_complete_.add_match_string(name);
 }
 
 bool ParameterAPI::set_api_variable(std::string name, std::string value) {
```

Comment:
These parameters are asking for a copy of the string memory. The older way would be to do:
set_api_variable(const std::string& name, const std::string& value) and operate on references instead of copies.

However you are right that string_view is likely the better choice here. string_view is like std::span in that it is designed to be a basically a pointer and a size, so it is like passing around a "const char*" but not depending on null termination for sizing so it is much safer.

For the uses I am seeing here all of the source memory are constants that should be in the static/code memory space and (ideally) wouldn't need to be loaded to ram (your milage may vary depending on MCU platform). But either way since the strings always exist you don't need to manage their memory and can just pass around the pointer (in std::string_view).